Mage.Sets/src/mage/cards/l/LoamcrafterFaun.java

```diff
@@ -0,0 +1,100 @@
+package mage.cards.l;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.common.delayed.ReflexiveTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.ReturnFromGraveyardToHandTargetEffect;
+import mage.constants.SubType;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterPermanentCard;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.common.TargetDiscard;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+
+/**
+ *
+ * @author Xanderhall
+ */
+public final class LoamcrafterFaun extends CardImpl {
+
+    public LoamcrafterFaun(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{G}");
+        
+        this.subtype.add(SubType.SATYR);
+        this.subtype.add(SubType.DRUID);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // When Loamcrafter Faun enters the battlefield, you may discard one or more land cards. When you do, return up to that many target nonland permanent cards from your graveyard to your hand.
+        this.addAbility(new EntersBattlefieldAbility(new LoamcrafterFaunDiscardEffect()));
+    }
+
+    private LoamcrafterFaun(final LoamcrafterFaun card) {
+        super(card);
+    }
+
+    @Override
+    public LoamcrafterFaun copy() {
+        return new LoamcrafterFaun(this);
+    }
+}
+
+class LoamcrafterFaunDiscardEffect extends OneShotEffect {
+
+    private static FilterCard filter = new FilterPermanentCard("nonland permanent(s) in your graveyard");
```

Comment:
is there a reason the text doesn't match the card? "nonland permanent cards from your graveyard"